src/cli/cmd/mod.rs

```diff
@@ -367,6 +382,109 @@ macro_rules! flakehub_url {
     }};
 }
 
+#[macro_export]
+macro_rules! path {
+    ($root:expr, $($segment:expr),+ $(,)?) => {{
+        let mut path = PathBuf::from($root);
+
+        $(
+            path.push($segment);
+        )+
+
+        path
+    }};
+}
+
+fn is_root_user() -> bool {
+    nix::unistd::getuid().is_root()
+}
+
+async fn nix_command(args: &[&str], sudo_if_necessary: bool) -> Result<(), FhError> {
+    command_exists("nix")?;
+
+    let use_sudo = sudo_if_necessary && !is_root_user();
+
+    let mut cmd = if use_sudo {
+        tracing::warn!(
+            "Current user is {} rather than root; running Nix command using sudo",
+            whoami::username()
+        );
+        tracing::debug!(
+            "Running: sudo nix --extra-experimental-features 'nix-command-flakes' {}",
+            args.join(" ")
+        );
+
+        let mut cmd = tokio::process::Command::new("sudo");
+        cmd.arg("nix");
+        cmd
+    } else {
+        tracing::debug!(
+            "Running: nix --extra-experimental-features 'nix-command-flakes' {}",
+            args.join(" ")
+        );
+
+        tokio::process::Command::new("nix")
+    };
+
+    let output = cmd
+        .args(["--extra-experimental-features", "nix-command flakes"])
+        .args(args)
+        .stdout(Stdio::inherit())
+        .stderr(Stdio::inherit())
+        .spawn()
+        .wrap_err("failed to spawn Nix command")?
+        .wait_with_output()
+        .await
+        .wrap_err("failed to wait for Nix command output")?;
+
+    if output.status.success() {
+        Ok(())
+    } else {
+        Err(FhError::FailedNixCommand)
+    }
+}
+
+fn parse_flake_output_ref(output_ref: &str) -> Result<FlakeOutputRef, FhError> {
+    // Ensures that users can use both forms:
+    // 1. https://flakehub/f/{org}/{project}/{version_req}#{output}
+    // 2. {org}/{project}/{version_req}#{output}
+    let output_ref = String::from(
+        output_ref
+            .strip_prefix("https://flakehub.com/f/")
```

Comment:
How does this work with dev instances?

Reply:
It's just stripping the prefix, so I suppose in dev I think you'd probably use org/project/...  or if you used prod URLs it'd get stripped down to org/project/.... But if you used staging prefixes it won't ... which ... what would you like? :)

Reply:
I would like it to work the same for both dev and prod URLs so that it's possible to test it outside of prod.

Reply:
I just boarded a plane but `api_addr` is globally available and could be used to construct the prefix

Reply:
Fixed up!